components/text_sanitize/strip_non_alphanumeric_or_ascii_characters_unittest.cc

```diff
@@ -0,0 +1,44 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include <string>
+
+#include "brave/components/text_sanitize/rs/src/lib.rs.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace text_sanitize {
+
+TEST(StripNonAlphanumericAsciiCharactersTest,
+     StripNonAlphanumericAsciiCharacters) {
+  const std::vector<std::tuple<std::string, std::string>> samples = {
```

Comment:
Can you also include a string with a null in the middle and some binary data stored as a string?

Reply:
Added tests. A binary data test showed that Rust doesn't recognize a NULL character ('\0') as an end of the string, so it was processed and replaced with whitespace.
So I decided to modify the strip function to terminate processing if a NULL character was encountered.

Reply:
This is a little strange. Like `std::string` rust strings have an explicit length and can contain internal nulls. Unlike (some?) C++ they are not explicitly null-terminated; there's a `CString` type you can convert to if you need that.

A null certainly isn't an alphanumeric character, but html documents can certainly contain them. What is the rationale for truncating at the first null rather than converting to whitespace like any other control character?

Reply:
I didn't know that NULL character is valid for HTML documents. So discarding logic is wrong. I will remove it.

Reply:
done